samples/07.advanced-web-chat-apps/e.sso-on-behalf-of-authentication/app/src/ui/WebChat/Notifications/BotSignInToast.js

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useContext, useEffect, useState, useRef } from 'react';
+import { hooks } from 'botframework-webchat';
+import { Spinner } from 'office-ui-fabric-react/lib/Spinner';
+
+import './index.css';
+import OAuthContext from '../../../oauth/Context';
+
+const { useActivities, useDismissNotification, usePostActivity, useSetNotification } = hooks;
+
+export const BotSignInToast = ({ notification }) => {
+  const {
+    data: { content },
+    id
+  } = notification;
+  const [authenticating, setAuthenticating] = useState();
+  const { acquireToken, getAccount, onSignIn } = useContext(OAuthContext);
+  const { connectionName, tokenExchangeResource: { id: oauthId, uri } = {} } = content;
+  const { current: invokeId } = useRef(
+    Math.random()
+      .toString(36)
+      .substr(2, 10)
+  );
+
+  const [activities] = useActivities();
+  const dismissNotification = useDismissNotification();
+  const postActivity = usePostActivity();
+  const setNotification = useSetNotification();
+
+  const exchangeToken = useCallback(
+    async resourceUri => {
+      const user = getAccount();
+      if (!user) {
+        await onSignIn();
+      }
+      const { accessToken } = await acquireToken({ scopes: [resourceUri] });
+      return accessToken;
+    },
+    [acquireToken, getAccount]
+  );
+
+  const handleDismiss = useCallback(() => {
+    dismissNotification(id);
+  }, [dismissNotification, id]);
+
+  useEffect(() => {
+    const invokeActivity = activities.find(
+      ({ channelData: { invokeId: activityInvokeId } = {} }) => invokeId === activityInvokeId
+    );
+    if (invokeActivity) {
+      const { channelData: { state } = {} } = invokeActivity;
+      if (state === 'send failed') {
+        dismissNotification(id);
+        setNotification({
+          id: 'traditionalbotauthentication',
+          data: { content },
+          level: 'error',
+          message: 'There was an error authenticating the bot.'
+        });
+      } else if (state === 'sent') {
+        dismissNotification(id);
+        setNotification({
+          id: 'signinsuccessful',
+          level: 'success',
+          message: 'The bot was authenticated successfully'
```

Comment:
```suggestion
          message: 'The bot was authenticated successfully.'
```

The language used in message looks weird to me.

Reply:
@corinagum thoughts on language?